ui/raidboss/timeline_parser.ts

```diff
@@ -461,6 +454,168 @@ export class TimelineParser {
     });
   }
 
+  private matchSyncNetRegex(
+    line: string,
+    lineNumber: number,
+    originalLine: string,
+    uniqueid: number,
+    seconds: number,
+    e: Event,
+  ) {
+    const commandMatch = regexes.syncNetRegex.exec(line);
+    if (!commandMatch || !commandMatch['groups'])
+      return line;
+
+    const syncCommand = commandMatch['groups'];
+    if (syncCommand.netRegexType === undefined || syncCommand.netRegex === undefined)
+      throw new UnreachableCode();
+
+    line = line.replace(syncCommand.netRegexType, '').trim();
+
+    const netRegexType = syncCommand.netRegexType;
+    if (!isLogDefinitionTypes(netRegexType)) {
+      this.errors.push({
+        lineNumber: lineNumber,
+        line: originalLine,
+        error: 'Invalid NetRegex type',
+      });
+      return line;
+    }
+
+    line = line.replace(syncCommand.netRegex, '').trim();
+
+    let params: unknown;
+    try {
+      params = JSON.parse(syncCommand.netRegex);
+    } catch (e) {
+      this.errors.push({
+        lineNumber: lineNumber,
+        line: originalLine,
+        error: 'Invalid NetRegex JSON',
+      });
+      return line;
+    }
+
+    if (!isObject(params) || !isValidNetParams(netRegexType, params)) {
+      this.errors.push({
+        lineNumber: lineNumber,
+        line: originalLine,
+        error: 'Invalid NetRegex arguments',
+      });
+      return line;
+    }
+    return this.buildRegexSync(
+      uniqueid,
+      `${netRegexType} ${syncCommand.netRegex}`,
+      Regexes.parse(this.GetReplacedSync(buildNetRegexForTrigger(netRegexType, params))),
```

Comment:
Now that we have parameters, can this use `translateRegexBuildParam` instead of `GetReplacedSync` here like `popup-text.ts` does? The main goal is that it knows which fields need to be translated rather than trying to translate the entire regex in one go.

I think you also might want to store the parameter object on the sync itself so that `find_missing_timeline_translations.ts` can just use that directly along with the return value from `translateRegexBuildParam` to know if a sync is fully translated. This would also let the `Translate` function in `timeline_parser.ts` use the same logic. (Sorry this is in a bunch of places.)

Reply:
Yeah, it being so spread out is why I didn't include it in this PR. I wanted to keep this as limited in scope as possible while not having the code be a mess.

Reply:
Ok, sounds good! Can you leave a TODO at least in these places that it should translate parameters and not regexes ideally?

Reply:
I think that `TODO` comment works?

Reply:
Looks good to me, thanks!